qiskit/circuit/library/pauli_product_measurement.py

```diff
@@ -0,0 +1,157 @@
+                The Pauli may include a phase of :math:`-1`, but not :math:`i` or :math:`-i`.
+            label: An optional label for the gate to display in circuit visualizations.
+                By default, the label is set to ``PPM(<pauli label>)``.
+
+        .. note::
+
+            While Paulis involving ``"I"``-terms are fully supported, it is recommended to remove
+            ``"I"``-terms from the Pauli when creating a ``PauliProductMeasurement`` instruction,
+            as this does not change the actual measurement but specifies the instruction over
+            a smaller set of qubits.
+
+        """
+
+        if not isinstance(pauli, Pauli):
+            raise CircuitError(
+                "A Pauli Product Measurement instruction can only be "
+                "instantiated from a Pauli object."
+            )
+
+        if _is_empty_pauli(pauli):
+            raise CircuitError(
+                "A Pauli Product Measurement instruction can not have an empty Pauli label."
+            )
+
+        if _is_identity_pauli(pauli):
+            raise CircuitError(
+                "A Pauli Product Measurement instruction can not have an all-'I' Pauli label."
+            )
+
+        if pauli.phase not in [0, 2]:
+            raise CircuitError(
+                "A Pauli Product Measurement instruction can not have a Pauli phase of i or -i."
+            )
+
+        num_qubits = len(pauli.z)
+        self._pauli_z = pauli.z
+        self._pauli_x = pauli.x
+        self._pauli_phase = pauli.phase
+
+        if label is None:
+            label = _get_default_label(pauli)
+
+        super().__init__(
+            name="pauli_product_measurement",
+            num_qubits=num_qubits,
+            num_clbits=1,
+            params=[],
+            label=label,
+        )
+
+    @classmethod
+    def _from_pauli_data(cls, z, x, phase, label):
+        """
+        Instantiates a PauliProductMeasurement isntruction from pauli data and label.
+        This function is used internally from within the rust code and from QPY
+        serialization.
+        """
+        return cls(Pauli((z, x, phase)), label)
+
+    def inverse(self, annotated=False):
+        """Prevents from calling ``inverse`` on a PauliProductMeasurement instruction."""
+        raise CircuitError("PauliProductMeasurement is not invertible.")
+
+    def __eq__(self, other):
+        if not isinstance(other, PauliProductMeasurement):
+            return False
+
+        if self.label != other.label:
+            return False
+
+        return (
+            np.all(self._pauli_z == other._pauli_z)
+            and np.all(self._pauli_x == other._pauli_x)
+            and (self._pauli_phase == other._pauli_phase)
+        )
+
+    def _define(self):
+        circuit = QuantumCircuit._from_circuit_data(
+            synth_pauli_product_measurement(self),
+            legacy_qubits=True,
+            name="ppm_circuit",
```

Comment:
from our offline discussions, maybe this? 🙂 
```suggestion
            name="def_ppm",
```